### PR TITLE
Add Joplin support

### PR DESCRIPTION
**Pull request by acron0**

### CODE REVIEW

File: resources/config.edn [hunk 1]
```diff
@@ -4,7 +4,10 @@
 
  :cassandra {:host #profile {:development #or [#env CASSANDRA "127.0.0.1"]
                              :production "cassandra-dcos-node.cassandra.dcos.mesos"}
-             :keyspace "witan_workspace"}
+             :keyspace "witan_workspace"
+             :joplin "joplin.edn"
+             :replication-factor #profile {:development 1
+                                           :production 1}}
```
Comment (otfrom) [inline resources/config.edn]: our replication factor in production should be 3 I think

File: src/migrators/cassandra/201606031548_init.clj [hunk 2]
```diff
@@ -0,0 +1,56 @@
+(ns migrators.cassandra.201606031548-init
```
Comment (otfrom) [inline src/migrators/cassandra/201606031548_init.clj:1]: we'll have to be careful of the ordering of these files when we have things coming in from multiple branches. Possibly a comment in the developer guidelines?

File: src/seeds/cassandra.clj [hunk 3]
```diff
@@ -0,0 +1,10 @@
+(ns seeds.cassandra
+  (:require [qbits.alia                 :as alia]
+            [qbits.hayt                 :as hayt]
+            [joplin.cassandra.database :refer [get-connection]]))
+
+(defn run [target & args]
+  (let [conn (get-connection (-> target :db :hosts)
+                             (-> target :db :keyspace))]
+    #_(alia/execute conn
```
Comment (otfrom) [inline src/seeds/cassandra.clj]: why the commented out procedure?

File: src/witan/workspace/components/cassandra.clj [hunk 4]
```diff
@@ -48,6 +59,15 @@
 
   component/Lifecycle
   (start [component]
+    (log/info "Bootstrapping Cassandra...")
+    (create-workspace! host keyspace replication-factor)
+    (let [joplin-config (jrepl/load-config (io/resource joplin))]
```
Comment (otfrom) [inline src/witan/workspace/components/cassandra.clj:64]: do we want to run this _every_ time we start a c\* connection or only on demand?

Comment (acron0) [inline src/witan/workspace/components/cassandra.clj:64]: Every time. Both workspace creation and migrations are idempotent.

Comment (otfrom) [inline src/witan/workspace/components/cassandra.clj:64]: OK. That's a big decision on how we will design these components. We can go with it for now.

**Comment by otfrom** [comment]
:shipit:

File: src/migrators/cassandra/201606031548_init.clj (hunk 2 above)
Comment (acron0) [inline src/migrators/cassandra/201606031548_init.clj:1]: Yeah, this is definitely true.

**Pull request merged**